ddp-workspace/projects/ddp-dsm-ui/src/app/shipping-search/shipping-search.component.ts

```diff
@@ -121,43 +120,39 @@ export class ShippingSearchComponent implements OnInit {
       ); // need to subscribe, otherwise it will not send!
   }
 
-  valueChanged( value: any, parameterName: string, kitRequest: KitRequest ): void {
-    let v;
-
-    if (typeof value === 'string') {
-      kitRequest[ parameterName ] = value;
-      v = value;
-    }
-    if (v != null) {
-      const realm: string = kitRequest.realm;
-      const patch1 = new PatchUtil(
-        kitRequest.dsmKitRequestId, this.role.userMail(),
-        {
-          name: parameterName,
-          value: v
-        }, null, 'dsmKitRequestId', kitRequest.dsmKitRequestId,
-        'kit', null, realm, kitRequest.participantId
-      );
-      const patch = patch1.getPatch();
-      this.currentPatchField = parameterName;
-      this.patch( patch );
-    }
+  patch(patch: any): Observable<any> {
+    return this.dsmService.patchParticipantRecord(JSON.stringify(patch));
   }
 
-  patch( patch: any ): void {
-    this.dsmService.patchParticipantRecord( JSON.stringify( patch ) ).subscribe( { // need to subscribe, otherwise it will not send!
-      next: data => {
-        this.currentPatchField = null;
-      },
-      error: err => {
-        if (err._body === Auth.AUTHENTICATION_ERROR) {
-          this.auth.logout();
-        }
-      }
-    } );
+//Triggers when user finishes typing
+valueChanged(value: any, parameterName: string, kitRequest: KitRequest): void {
+  if (typeof value === 'string') {
+    kitRequest[parameterName] = value;
   }
+}
 
-  isPatchedCurrently( field: string ): boolean {
-    return this.currentPatchField === field;
-  }
+saveCompleted(): void{
+  this.currentPatchField = null;
 }
+
+  //Fires when user clicks "Save Date" button. Returns true
+  //if the promise succeeds and the server successfukky writes and
+  //returns and error if the server write fails.
+  saveDate(kitRequest: KitRequest): Observable<boolean> { 
+      return defer(() => {
```

Comment:
**# Optional**
Why did we need `defer()` here? 
As far as I know, it's used mostly in 2 scenarios:
- If we want to execute code on subscribe that will be eagerly executed, like Promise.
- If we want to execute an Observable call, but with some side effects, that should happen only when subscribed (This probably can be handled in `pipe()` as well)

Please see the [discussion](https://stackoverflow.com/questions/38764578/rxjs-understanding-defer).

Maybe we could simply return the `this.patch()`, which itself returns an observable that won't be executed eagerly. 

P.S. Maybe we've got a scenario, where it's necessary to use `defer()` and I didn't find it yet, please let me know.

Reply:
This is left over from an attempt to solve the infinite loop issue, I can remove it as it is not needed.